File: Beecrowd/Beginner/2003 - 2176/2163 - The Force Awakens/C++/2163_TheForceAwakens.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int rows, columns;
    cin >> rows >> columns;

    int matrix[rows][columns];

    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < columns; j++) {
            cin >> matrix[i][j];
        }
    }

    int count = 0;
    
    for (int i = 1; i < rows - 1; i++) {
        for (int j = 1; j < columns - 1; j++) {
            if (matrix[i][j] == 42 && matrix[i - 1][j - 1] == 7 &&
                matrix[i - 1][j] == 7 && matrix[i - 1][j + 1] == 7 &&
                matrix[i][j - 1] == 7 && matrix[i][j + 1] == 7 &&
                matrix[i + 1][j - 1] == 7 && matrix[i + 1][j] == 7 &&
                matrix[i + 1][j + 1] == 7) {
                cout << i + 1 << " " << j + 1 << endl;
                count++;
            }
        }
    }

    if (count == 0) {
        cout << "0 0" << endl;
    }

    return 0;
}

File: Beecrowd/Beginner/2003 - 2176/2163 - The Force Awakens/Python/2163_TheForceAwakens.py
rows, columns = map(int, input().split())

matrix = []
for _ in range(rows):
    row = list(map(int, input().split()))
    matrix.append(row)

count = 0

for i in range(1, rows - 1):
    for j in range(1, columns - 1):
        if matrix[i][j] == 42:
            if matrix[i - 1][j - 1] == 7 and matrix[i - 1][j] == 7 and matrix[i - 1][j + 1] == 7:
                if matrix[i][j - 1] == 7 and matrix[i][j + 1] == 7:
                    if matrix[i + 1][j - 1] == 7 and matrix[i + 1][j] == 7 and matrix[i + 1][j + 1] == 7:
                        print(i + 1, j + 1)
                        count += 1
                        break
    if count > 0:
        break

if count == 0:
    print("0 0")
